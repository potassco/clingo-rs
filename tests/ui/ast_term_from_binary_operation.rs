// ast_term_from_binary_operation
use clingo::ast::symbolic_term;
use clingo::ast::Location;
use clingo::*;
fn main() {
    let loc = Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term1 = symbolic_term(&loc, &sym).unwrap();
    let term2 = symbolic_term(&loc, &sym).unwrap();
    let xor = ast::BinaryOperator::Xor;
    let op = ast::binary_operation(&loc, xor, term1, term2).unwrap();
    let term3: ast::Term = op.into();
    drop(op);
    let _end = term3;
}
