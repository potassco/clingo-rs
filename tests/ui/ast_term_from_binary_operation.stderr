error[E0382]: use of moved value: `op`
  --> tests/ui/ast_term_from_binary_operation.rs:13:10
   |
11 |     let op = ast::binary_operation(&loc, xor, term1, term2).unwrap();
   |         -- move occurs because `op` has type `clingo::ast::BinaryOperation<'_>`, which does not implement the `Copy` trait
12 |     let term3: ast::Term = op.into();
   |                               ------ `op` moved due to this method call
13 |     drop(op);
   |          ^^ value used here after move
   |
note: `into` takes ownership of the receiver `self`, which moves `op`
  --> $RUST/core/src/convert/mod.rs
   |
   |     fn into(self) -> T;
   |             ^^^^
help: you can `clone` the value and consume it, but this might not be your desired behavior
   |
12 |     let term3: ast::Term = op.clone().into();
   |                               ++++++++
